edgelet/iotedge/src/check/mod.rs

```diff
@@ -334,190 +343,306 @@ impl Check {
-                                },
-                                additional_info: check.get_json(),
-                            },
-                        );
+        // run the aziot checks first, as certain bits of `additional_info` from
+        // aziot are required to run iotedge checks. e.g: the "iothub_hostname".
+        {
+            fn to_check_result(res: CheckResultSerializable) -> CheckResult {
+                fn vec_to_err(mut v: Vec<String>) -> failure::Error {
+                    let mut err =
+                        failure::err_msg(v.pop().expect("errors always have at least one source"));
+                    while let Some(s) = v.pop() {
+                        err = err.context(s).into();
+                    }
+                    err
+                }
+
+                match res {
+                    CheckResultSerializable::Ok => CheckResult::Ok,
+                    CheckResultSerializable::Warning { details } => {
+                        CheckResult::Warning(vec_to_err(details))
+                    }
+                    CheckResultSerializable::Ignored => CheckResult::Ignored,
+                    CheckResultSerializable::Skipped => CheckResult::Skipped,
+                    CheckResultSerializable::Fatal { details } => {
+                        CheckResult::Fatal(vec_to_err(details))
+                    }
+                    CheckResultSerializable::Error { details } => {
+                        CheckResult::Failed(vec_to_err(details))
+                    }
+                }
+            }
 
-                        stdout.write_error(|stdout| {
-                            writeln!(stdout, "\u{00d7} {} - Error", check_name)?;
+            let mut aziot_check = std::process::Command::new(&self.aziot_bin);
+            aziot_check
+                .arg("check")
+                .arg("-o=json-stream")
+                .stdout(std::process::Stdio::piped());
 
-                            let message = err.to_string();
+            if let Some(iothub_hostname) = &self.iothub_hostname {
+                aziot_check.arg("--iothub-hostname").arg(iothub_hostname);
+            }
 
-                            write_lines(stdout, "    ", "    ", message.lines())?;
+            if !self.dont_run.is_empty() {
+                aziot_check
+                    .arg("--dont-run")
+                    .arg(self.dont_run.iter().cloned().collect::<Vec<_>>().join(" "));
+            }
 
-                            if self.verbose {
-                                for cause in err.iter_causes() {
-                                    write_lines(
-                                        stdout,
-                                        "        caused by: ",
-                                        "                   ",
-                                        cause.to_string().lines(),
-                                    )?;
+            match aziot_check.spawn() {
+                Err(err) => {
```

Comment:
Put the `Err` arm after the `Ok` arm please, to be consistent with the rest of the codebase.